crates/astria-conductor/src/executor/mod.rs

```diff
@@ -149,39 +202,56 @@ impl Executor {
                 }
 
                 cmd = self.cmd_rx.recv() => {
-                    let Some(cmd) = cmd else {
-                        error!("cmd channel closed unexpectedly; shutting down");
+                    if let Err(e) = self.handle_executor_command(cmd).await {
+                        error!(error.message = %e, "failed to handle executor command, breaking from executor loop");
                         break;
-                    };
-                    match cmd {
-                        ExecutorCommand::FromSequencer {
-                            block,
-                        } => {
-                            let height = block.header().height.value();
-                            let block_subset =
-                                SequencerBlockSubset::from_sequencer_block_data(*block, &self.chain_id);
-
-                            if let Err(e) = self.execute_block(block_subset).await {
-                                error!(
-                                    sequencer_block_height = height,
-                                    error = ?e,
-                                    "failed to execute block"
-                                );
-                            }
-                        }
+                    }
+                }
+            );
+
+            #[cfg(feature = "optimism")]
```

Comment:
It's going to make it very hard to reason about how conductor works if we start using conditional compilation like this.

Reply:
yeah, the two select blocks are the same apart from the `stream.next()` block for the optimism feature. i had to write it this way since `select!` doesn't allow conditional compilation in it: https://github.com/tokio-rs/tokio/issues/3974

i'm not sure of a better method for the optimism integration than a feature, since the optimism feature would be used only by a rollup which is deriving from ETH, so if a rollup isn't deriving from ETH there's no reason to have any optimism stuff in the compiled code. thoughts?